src/Device/Driver/LX_EOS/LXEosDeclare.cpp

```diff
@@ -0,0 +1,218 @@
+                               sizeof(data));
+
+  return WriteAndWaitForACK(message, env);
+}
+
+bool
+LXEosDevice::SendObsZone(uint8_t tp_nr,
+                         const Declaration& declaration,
+                         OperationEnvironment& env)
+{
+  if (tp_nr < 1 || tp_nr > declaration.Size())
+    return false;
+
+  EosObsZoneStruct data;
+  const auto& turnpoint = declaration.turnpoints.at(tp_nr - 1);
+
+  data.uiTpNr = tp_nr;
+  data.uiR1 = ToLE32(turnpoint.radius);
+
+  if (tp_nr == 1)                       // Start
+    data.uiDirection = 2;               // Next
+  else if (tp_nr == declaration.Size()) // Finish
+    data.uiDirection = 3;               // Previous
+  else                                  // Turnpoint
+    data.uiDirection = 0;               // Symmetric
+
+  data.bIsLine = (turnpoint.shape == Declaration::TurnPoint::LINE) ? 1 : 0;
+
+  /* We cannot be sure about the format of float on the platform, hardcoded
+   * bytes are used for the angles */
+
+  // Bytes of float values for 45, 90 and 180 degree angles in radians
+  uint8_t angle180[] = { 0xdb, 0x0f, 0x49, 0x40 };
+  uint8_t angle90[] = { 0xdb, 0x0f, 0xc9, 0x3f };
+  uint8_t angle45[] = { 0xdb, 0x0f, 0x49, 0x3f };
+
+  // Init A21 and A2 to zero
+  memset(&data.fA21, 0, 4);
+  memset(&data.fA2, 0, 4);
+
+  switch (turnpoint.shape) {
+    case Declaration::TurnPoint::LINE:
+      memcpy(&data.fA1, &angle90, 4);
+      break;
+    case Declaration::TurnPoint::SECTOR:
+      memcpy(&data.fA1, &angle45, 4);
+      break;
+    case Declaration::TurnPoint::CYLINDER:
+      memcpy(&data.fA1, &angle180, 4);
+      break;
+    case Declaration::TurnPoint::DAEC_KEYHOLE:
+      memcpy(&data.fA1, &angle45, 4);
+      memcpy(&data.fA2, &angle180, 4);
+      data.uiR2 = ToLE32((uint32_t)500);
+      break;
+    default:
+      memcpy(&data.fA1, &angle180, 4); // Cylinder
+  }
+
+  /* We cannot use predefined bytes for elevation (it can be any value)
+   * Check if the float format on this platform matches the expected one
+   * by comparing with 4 bytes oof known-good value */
+  float test_float = 123.45e33;
+  uint8_t expected_bytes[] = { 0x70, 0x34, 0xbe, 0x79 };
+  if (memcmp(&test_float, &expected_bytes, 4) == 0) {
+    // Yes
+    data.fElevation = declaration.GetWaypoint(tp_nr - 1).GetElevationOrZero();
+  } else {
+    // No. Maybe the bytes are swapped?
+    uint32_t test_float_bytes = 0;
+    memcpy(&test_float_bytes, &test_float, 4);
+    test_float_bytes = ByteSwap32(test_float_bytes);
+    if (memcmp(&test_float_bytes, &expected_bytes, 4) == 0) {
+      // Use swapped bytes
+      float elev = declaration.GetWaypoint(tp_nr - 1).GetElevationOrZero();
+      uint32_t elev_bytes = 0;
+      memcpy(&elev_bytes, &elev, 4);
+      elev_bytes = ByteSwap32(elev_bytes);
+      memcpy(&data.fElevation, &elev_bytes, 4);
+    } else {
+      // Set zero if the format is something exotic
+      memset(&data.fElevation, 0, 4);
+    }
+  }
```

Comment:
All of this is ... awful. This adds runtime overhead by forcing the compiler to generate memcmp() calls, even on CPUs where nothing needs to be changed. All of these checks should be at compile-time and generate only runtime code if runtime code is really necessary. And it should be encapsulated in a generic class that can be reused everywhere. Solve this once, properly, and then reuse this class everywhere you need a float in a specific bit format with a specific byte order.